tfhe/src/core_crypto/gpu/mod.rs

```diff
@@ -341,19 +384,30 @@ pub unsafe fn convert_lwe_programmable_bootstrap_key_async<T: UnsignedInteger>(
     l_gadget: DecompositionLevelCount,
     polynomial_size: PolynomialSize,
 ) {
-    let size = std::mem::size_of_val(src);
     for (i, &stream_ptr) in streams.ptr.iter().enumerate() {
-        assert_eq!(dest.len() * std::mem::size_of::<T>(), size);
-        cuda_convert_lwe_programmable_bootstrap_key_64(
-            stream_ptr,
-            streams.gpu_indexes[i].get(),
-            dest.as_mut_c_ptr(i as u32),
-            src.as_ptr().cast(),
-            input_lwe_dim.0 as u32,
-            glwe_dim.0 as u32,
-            l_gadget.0 as u32,
-            polynomial_size.0 as u32,
-        );
+        if size_of::<T>() == 16 {
```

Comment:
I'm not sure that this is the best way to do this in Rust. @IceTDrinker maybe you could give us some advice here? We have convert_lwe_programmable_bootstrap_key_async that should call the u128 entry point or the u64 one depending on the type T. Is using an if in this way alright?

Reply:
Actually it is pretty convenient this way, we only have 1 entry point for the GPU bootstrap in Rust, regardless whether it's u64 or u128, I quite like it.